app/services/foreman_discovery/host_converter.rb

```diff
@@ -52,14 +52,25 @@ def self.unused_ip_for_subnet(subnet, mac, existing_ip)
     end
   end
 
+  def self.ip_for_subnet(subnet, mac, ip) 
+    if ip
+      ipam = subnet.unused_ip(mac)
+
+      return ip if ipam.subnet_range.include?(ip)
```

Comment:
For the record, this scans the whole range, which in case of IPv6 is huge. Use IPAddr.include? instead, better to improve Foreman core IPAM::Base implementation.